File: api/economy/definitions/economy.ts
export type AccountType = "cash" | "bank";

export interface ServerEconomyProvider {
    getPlayerBalance(playerId: number, account: AccountType): number | undefined;
    setPlayerBalance(playerId: number, account: AccountType, amount: number): boolean;
    addPlayerBalance(playerId: number, account: AccountType, amount: number): boolean;
    removePlayerBalance(playerId: number, account: AccountType, amount: number): boolean;
    playerHasBalance(playerId: number, account: AccountType): boolean;
}

File: api/economy/definitions/economy.lua
---@alias AccountType "cash" | "bank"

---@class ServerEconomyProvider
---@field getPlayerBalance fun(playerId: integer, account: AccountType): number | nil
---@field setPlayerBalance fun(playerId: integer, account: AccountType, amount: number): boolean
---@field addPlayerBalance fun(playerId: integer, account: AccountType, amount: number): boolean
---@field removePlayerBalance fun(playerId: integer, account: AccountType, amount: number): boolean
---@field playerHasBalance fun(playerId: number, account: AccountType, amount: number): boolean
